File: EJERCICIOS/SEGUNDO PARCIAL/CONSULTAS/Consulta01/Shellshort/shellshort.java
public class ShellSort {
    public static void shellSort(int[] arr) {
        int n = arr.length;
        int gap = n / 2;

        while (gap > 0) {
            for (int i = gap; i < n; i++) {
                int temp = arr[i];
                int j = i;
                while (j >= gap && arr[j - gap] > temp) {
                    arr[j] = arr[j - gap];
                    j -= gap;
                }
                arr[j] = temp;
            }
            gap /= 2;
        }
    }

    public static void main(String[] args) {
        int[] arr = {5, 2, 8, 12, 1};

        shellSort(arr);

        System.out.print("Array ordenado: ");
        for (int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();
    }
}

File: EJERCICIOS/SEGUNDO PARCIAL/CONSULTAS/Consulta01/Shellshort/shellshort.cpp
#include <iostream>
using namespace std;

void shellSort(int arr[], int n) {
    // Definir las brechas iniciales
    for (int gap = n / 2; gap > 0; gap /= 2) {
        // Realizar la ordenación por inserción con la brecha actual
        for (int i = gap; i < n; i++) {
            int temp = arr[i];
            int j;
            for (j = i; j >= gap && arr[j - gap] > temp; j -= gap) {
                arr[j] = arr[j - gap];
            }
            arr[j] = temp;
        }
    }
}

int main() {
    int arr[] = {5, 2, 8, 12, 1};
    int n = sizeof(arr) / sizeof(arr[0]);

    shellSort(arr, n);

    cout << "Array ordenado: ";
    for (int i = 0; i < n; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;

    return 0;
}
